pkg/api/validation/validation.go

```diff
@@ -2045,3 +2052,21 @@ func ValidateLoadBalancerStatus(status *api.LoadBalancerStatus) errs.ValidationE
 	}
 	return allErrs
 }
+
+// ValidateEvent makes sure that the event makes sense.
+func ValidateEvent(event *api.Event) errs.ValidationErrorList {
+	allErrs := errs.ValidationErrorList{}
+
+	requiresNamespace := true
+	if event.InvolvedObject.Kind == "Node" {
+		// TODO: There is no namespace required for node.
+		requiresNamespace = false
+	}
+	allErrs = append(allErrs, ValidateObjectMeta(&event.ObjectMeta, requiresNamespace, ValidateEventName).Prefix("metadata")...)
```

Comment:
This will enforce presence or absence of namespace on the event, which wasn't previously done. I don't know all the places events are used… is node really the only root scope resource we'll allow an unnamespaced event on?